### PR TITLE
feat: Switch component 

### LINKED ISSUE #682: [Feature]: <Switch/>
### What problem does this feature solve?

<Switch/> has been requested by a lot of users. Tremor team is also reworking their entire block section where they find a need for this component.

### What does the proposed API look like?

Please use [headless' switch](https://headlessui.com/react/switch) for the underlying logic. See styling in the image below.

Explicit props:

- checked?: (value: string) => void;
- onChange?: (value: string) => void;
- color?: string // only for isChecked color
- id?: string
- name?: string
- error?: boolean
- errorMessage?: string
- disabled?: string

_gh_switch

### PR DESCRIPTION
**Pull request by AlexRousseau92**
**Description**

Add new component : Switch 

**Related issue(s)**

#682

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**How has This been tested?**

**Screenshots (if appropriate):**

Capture d’écran 2023-10-17 à 21 15 29

**The PR fulfills these requirements:**

- [x] It's submitted to the `main` branch
- [x] When resolving a specific issue, it's referenced in the related issue section above
- [x] My change requires a change to the documentation. (Managed by Tremor Team)
- [x] I have added tests to cover my changes
- [x] Check the ["Allow edits from maintainers" option](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) while creating your PR.
- [x] Add refs #XXX or fixes #XXX to the related issue section if your PR refers to or fixes an issue.
- [x] By contributing to Tremor, you confirm that you have read and agreed to Tremor's [CONTRIBUTING.md](https://github.com/tremorlabs/tremor/blob/main/CONTRIBUTING.md) guideline.

### CODE REVIEW

File: src/components/input-elements/Switch/Switch.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import { Switch as HeadlessSwitch } from "@headlessui/react";
+import { useInternalState } from "hooks";
+import { Color, colorPalette, getColorClassNames, makeClassName, tremorTwMerge } from "lib";
+
+import React, { useState } from "react";
+import { twMerge } from "tailwind-merge";
```
Comment (severinlandolt) [inline src/components/input-elements/Switch/Switch.tsx]: Is there a reason why twMerge is used and not tremorTwMerge?

Comment (AlexRousseau92) [reply src/components/input-elements/Switch/Switch.tsx]: Hi ! No it's a mistake

Comment (severinlandolt) [reply src/components/input-elements/Switch/Switch.tsx]: If you could push a fix, I am happy to merge it into beta :)

Comment (AlexRousseau92) [reply src/components/input-elements/Switch/Switch.tsx]: Yes of course ! I will :)

Comment (severinlandolt) [reply src/components/input-elements/Switch/Switch.tsx]: Thank you!

**Pull request merged**